core/services/AuthService.js

```diff
@@ -1,8 +1,26 @@
 const bcrypt = require("bcrypt");
 const config = require("config");
+const jwt = require("jsonwebtoken");
 
 module.exports = {
   encryptPassword: password => bcrypt.hashSync(password, config.get("bcryptSaltRounds")),
 
-  checkPassword: (password, passwordHash) => bcrypt.compareSync(password, passwordHash)
+  checkPassword: (password, passwordHash) => bcrypt.compareSync(password, passwordHash),
+
+  getOperatorByCredentials: async (email, password) => {
+    let operator = await yaxys.db.findOne ("operator", {email: email}, {});
+    if (operator && AuthService.checkPassword(password, operator.passwordHash)) {
+      return operator;
+    }
+    throw new Error("No such operator or wrong password provided");
+  },
+
+  generateToken: (operator) => jwt.sign(operator, config.get("jwt.secret"), {expiresIn: config.get("jwt.lifetime")}),
```

Comment:
By doing this, you make all the information about the operator available to the client – since you later send this token through the cookie. All the information including passwordHash. That is bad, but still not critical – since the hash can't be decoded into the password. But – if we later decide to put some more secret information into the operator model, we'll apparently forget about this moment an create potential security flaw.

So it's better to prevent it right now by putting only specified operator attributes into the token using lodash `pick` method:
```
jwt.sign(_.pick(operator, "id", "email")
```
I we later need some additional  info from operator, we'd better add it explicitly – that is much safer.

Reply:
And, since you have single argument and don't have square brackets around the function body, you can remove brackets surrounding argument: `operator => jwt...`